### PR TITLE
[CDU] Bugfixes to INIT and FUEL PRED pages

### LINKED ISSUE #2515: [BUG] EFOB dissapears in the middle of the flight with no reason
**Mod Version**
Stable release 0.5

**Describe the bug**
In the FUEL PREDICTION PAGE , Suddenly in the middle of the flight some paremetres in the same page and the EFOB dissapears

**To Reproduce**

1.
2.
3.

**Expected behavior**
Work as intended in the whole flight

**Actual behavior**
It just dissapears suddenly 

image

**References**
image

**Additional context**

Was this working before/when did the issue start occurring?
No , it was working a week ago

Is this a problem in the vanilla unmodded game?
No

Discord username (if different from GitHub):

### PR DESCRIPTION
**Pull request by Lucky38i**
Fixes #2515
Fixes #2507 

## Summary of Changes

- [x] Merge alternate and trip coefficient tables 
- [x] Fix error message logic
- [x] add EFOB to FPLAN
- [x] Add logic to check entered reserved weight isn't beyond percent limt
- [x] Try to re-work pred-failure or remove it

Discord username (if different from GitHub): Lucky38i

## Testing instructions

- Test trips between 20NM - 3100NM and FL100 - FL 390, verify you get predictions for anything in those parameters
- Check EFOB shows in FPLAN page, AFTER doing your INIT B, verify it's dashed before
- Try to enter a reserve weight that's 15% above the trip weight, you should get dashes results signify out of range, no error prints.
- Try to test any other bugs you're aware about regarding INIT and FUEL PRED.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_FuelPred.js [hunk 1]
```diff
@@ -207,26 +207,14 @@ class A32NX_FuelPred {
...
-                    return (Math.round(math.sum(math.dotMultiply(altCorrectionsCoeff,mmOfDistFL))));
-            }
+        switch (computation) {
+            case this.computations.FUEL:
+                const deviation = 0;
+                return (Math.round(math.sum(math.dotMultiply((!alternate ? fuelConsumedCoeff : altFuelConsumedCoef) ,mmOfDistFL)))) + deviation;
+            case this.computations.TIME:
+                return (Math.round(math.sum(math.dotMultiply((!alternate ? timeCoeff : altTimeCoef),mmOfDistFL))));
+            case this.computations.CORRECTIONS:
+                return (Math.round(math.sum(math.dotMultiply((!alternate ? correctionsCoef : altCorrectionsCoeff),mmOfDistFL))));
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_FuelPred.js:217]:
```suggestion
                return (Math.round(math.sum(math.dotMultiply((alternate ? altFuelConsumedCoef : fuelConsumedCoeff ) ,mmOfDistFL)))) + deviation;
            case this.computations.TIME:
                return (Math.round(math.sum(math.dotMultiply((alternate ? altTimeCoef : timeCoeff ),mmOfDistFL))));
            case this.computations.CORRECTIONS:
                return (Math.round(math.sum(math.dotMultiply((alternate ? altCorrectionsCoeff : correctionsCoef ),mmOfDistFL))));
```

Comment (MisterChocker) [reply A32NX/html_ui/Pages/A32NX_Core/A32NX_FuelPred.js:217]: I'd say that makes stuff easier to read and there is no brain fuck with true being false and false being true

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_FuelPred.js [hunk 2]
```diff
@@ -207,26 +207,14 @@ class A32NX_FuelPred {
...
-                    return (Math.round(math.sum(math.dotMultiply(altFuelConsumedCoef,mmOfDistFL)))) + deviation;
-                case this.computations.TIME:
-                    return (Math.round(math.sum(math.dotMultiply(altTimeCoef,mmOfDistFL))));
-                case this.computations.CORRECTIONS:
-                    return (Math.round(math.sum(math.dotMultiply(altCorrectionsCoeff,mmOfDistFL))));
-            }
+        switch (computation) {
+            case this.computations.FUEL:
+                const deviation = 0;
+                return (Math.round(math.sum(math.dotMultiply((alternate ? altFuelConsumedCoef : fuelConsumedCoeff), mmOfDistFL)))) + deviation;
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_FuelPred.js:213]: I'd like to understand why assign a const 0 right before and add 0 to the return?

**Review by Lucky38i — COMMENTED** [review]
I Can't seem to delete this comment, ignore

**Review by MisterChocker — APPROVED** [review]
LGTM

**Comment by Road-to-Hell9** [comment]
Category: QA Tester
Name: Road_to_Hell9 #1111
Date of testing: 22/12/2020
Version of the sim: Newest **22/12/2020**
PR Tested: #2604 
Object of test: [CDU] Bugfixes to INIT and FUEL PRED pages

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1.Cold start at EIDW
2.Set up a flight plan 
3.Entered values
4.

Flight notes (If required): Did Worked as reported

Issues (If Any): 
Overall Rating: To be improved
Conclusions of the testing: The Fuel Pred and Init B page just display Untiltled 
Screenshot (136)

**Comment by Road-to-Hell9** [comment]
Why did you delete my comment
It still doesn't work

**Comment by Lucky38i** [comment]
> Why did you delete my comment
> 
> It still doesn't work

Your QA was unhelpful and didn't provide anything in the name of quality assurance, I've contact you on discord

**Comment by Highflyer2010** [comment]
Category: QA Tester
Name: Discord User Highflyer2010#8932
Date of testing: Current date 23/12/2020
Version of the sim: 1.12.13.0
PR Tested: #2604
Object of test: [CDU] Bugfixes to INIT and FUEL PRED pages

Testing technique: C&D startup to test MCDU pages

Steps you've made to test the Pull Request:

1. Loaded Sim and loaded Aircraft C&D
 
2. Spawned at LSZH at a gate

3. Setup Aircraft , Ext Pwr, Batteries ect...
 
Flight notes (If required):N/A

Issues (If Any): Switched on EXT PWR then batteries & all Fuel Pumps 2 are OFF and won't turn off

Overall Rating: Poor

Conclusions of the testing: I was testing MCDU pages but ran into a problem on the overhead 

Video:  https://files.fm/u/bghr36jav

**Comment by Lucky38i** [comment]
@Highflyer2010 Can you please resubmit/edit your QA? the PR works now.

**Comment by BritishAvgeek** [comment]
Category: QA Tester
Name: BritishAvgeek#1436
Date of testing: 23/12/2020
Version of the sim: 1.12.13.0
PR Tested: #2604
Object of test: Bugfixed Init and Fuel Pred page

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. Cold and Dark at EGBB
2. Tried to build Init page for EGBB - LFPO, got no company routes so built a basic flight plan... headed into Init B
3. I added ZFW/ZFWCG and then clicked the Init Fuel Prediction button... didn't do anything - waited 30 seconds, tried again, didn't do anything
4. I then typed in my block fuel anyway to see if it would predict based on manual input - it did for 1 second then it blanked the entire MCDU to "untitled"
5. Had to exit to main menu and respawn to reset the MCDU

Flight notes (If required):

Issues (If Any): As commented
Overall Rating: To be improved
Conclusions of the testing: The moment I tried to initiate a fuel prediction with the button to self-calculate it didn't do anything. I then input my block fuel manually and it crashed the MCDU as shown below and all buttons on the MCDU stopped working:
image

**Comment by sidnov** [comment]
Category: QA Tester
Name: sidnov#8337
Date of testing: 23/12/2020
Version of the sim: 1.12.13.0
PR Tested: #2604
Object of test: Test the bugfixes to INIT and FUEL PRED pages

Testing technique: Spawn at gates cold and dark

Steps you've made to test the Pull Request:
1. Spawned at VABB Gate 5, C/D
2. Started Batteries, External GPU, Set ADIRS to NAV, Set lights.
3. Filled INIT A page with route, cost index, and CRZ level
4. In INIT B, clicked on the ZFW/ZFWCG and set the values. Proceeded to set Block Fuel
5. INIT B page crashed with "UNTITLED" on setting block fuel. The INIT B page was unrecoverable.
6. Other CDU pages worked fine.

Issues (If Any): UNTITLED bug as mentioned above.
Overall Rating: Needs improvement
Conclusions of the testing: As noticed in the other tests as well, there is an issue with the current build resulting in the INIT B page not working.

INIT Bug

**Comment by Highflyer2010** [comment]
Guy's when tested did use input sid and star before pressing the ZFW/ZFWCG and then fuel planning because on the current DEV as of now i've in put sid and star and now fuel calcations planning button is working with no "UNTITLED" page showing

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js [hunk 3]
```diff
@@ -1864,7 +1852,13 @@ class FMCMainDisplay extends BaseAirliners {
             if (isFinite(rteRsvWeight)) {
                 this._routeReservedWeight = rteRsvWeight;
                 this._routeReservedPercent = 0;
-                return true;
+                if (this.isRteRsvPercentInRange(this.getRouteReservedPercent())) { //Bit of a hacky method due previous tight coupling of weight and percentage calculations
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js:1855]:
```suggestion
                if (this.isRteRsvPercentInRange(this.getRouteReservedPercent())) { // Bit of a hacky method due previous tight coupling of weight and percentage calculations
```

**Comment by BritishAvgeek** [comment]
Original Test edited to reflect bug fixes/changes with a brief summary of Test 2 - all worked well

**Pull request merged**